File: VB/CodeBehind/EFCore/ServerMode/Issues/IssuesContextInitializer.vb
Imports System
Imports System.Linq

Namespace Issues
    Public Module IssuesContextInitializer
        Public Sub Seed()
            Dim context = New IssuesContext()
            Dim users = OutlookDataGenerator.Users.[Select](Function(x)
                                                                Dim split = x.Split(" "c)
                                                                Return New User() With {
                                                                    .FirstName = split(0),
                                                                    .LastName = split(1)
                                                                }
                                                            End Function).ToArray()
            context.Users.AddRange(users)
            context.SaveChanges()
            Dim rnd = New Random(0)
            Dim issues = Enumerable.Range(0, 1000).[Select](Function(i) New Issue() With {
                .Subject = OutlookDataGenerator.GetSubject(),
                .UserId = users(rnd.Next(users.Length)).Id,
                .Created = Date.Today.AddDays(-rnd.Next(30)),
                .Priority = OutlookDataGenerator.GetPriority(),
                .Votes = rnd.Next(100)
            }).ToArray()
            context.Issues.AddRange(issues)
            context.SaveChanges()
        End Sub
    End Module
End Namespace

File: CS/ViewModel/EntityFramework/ServerMode/Issues/IssuesContextInitializer.cs
using System;
using System.Data.Entity;
using System.Linq;

namespace EntityFrameworkIssues.Issues {
    public class IssuesContextInitializer
        : DropCreateDatabaseIfModelChanges<IssuesContext> {
        //: DropCreateDatabaseAlways<IssuesContext> { 

        public static void ResetData() {
            using(var context = new IssuesContext()) {
                context.Users.Load();
                context.Users.RemoveRange(context.Users);
                context.SaveChanges();
                CreateData(context);
            }
        }

        protected override void Seed(IssuesContext context) {
            base.Seed(context);
            CreateData(context);
        }

        static void CreateData(IssuesContext context) {
            var users = OutlookDataGenerator.Users
                .Select(x => {
                    var split = x.Split(' ');
                    return new User() {
                        FirstName = split[0],
                        LastName = split[1]
                    };
                })
                .ToArray();
            context.Users.AddRange(users);
            context.SaveChanges();

            var rnd = new Random(0);
            var issues = Enumerable.Range(0, 1000)
                .Select(i => new Issue() {
                    Subject = OutlookDataGenerator.GetSubject(),
                    UserId = users[rnd.Next(users.Length)].Id,
                    Created = DateTime.Today.AddDays(-rnd.Next(30)),
                    Priority = OutlookDataGenerator.GetPriority(),
                    Votes = rnd.Next(100),
                })
                .ToArray();
            context.Issues.AddRange(issues);

            context.SaveChanges();
        }
    }
}
